### PR TITLE
Support IAccessible2 "mark" text attribute.

### PR DESCRIPTION
**Pull request by jcsteh**
### Link to issue number:
None.

### Summary of the issue:
Text fragments is a new web concept that allows you to link directly to a specified piece of arbitrary text on a page without needing a predefined id. For example:
https://www.jantrid.net/#:~:text=browsers%20were%20much%20simpler
As well as scrolling to the element containing the text, the browser will highlight the text. Because this could span a partial object, this can't be exposed using IA2_ROLE_MARK. Instead, similar to spelling errors, this is exposed using an IAccessible2 text attribute, specifically the new attribute mark:true.

This will also be used eventually to support CSS custom highlights.

### Description of user facing changes
In Mozilla Firefox, NVDA will report the highlighted text when a URL containing a text fragment is visited.

### Description of development approach
NVDA already supports the "marked" attribute on FormatFields. To be consistent with IA2_ROLE_MARK, it was decided to name the IA2 attribute "mark" instead of "marked". This PR simply maps the attribute appropriately in `normalizeIA2TextFormatField`.

### Testing strategy:
1. Opened this test case in Firefox Nightly (this isn't enabled in Firefox release yet):
    `data:text/html,<p>The first phrase.</p><p>The second <b>phrase.</b></p>#:~:text=second phrase`
2. Moved to the second line.
3. Verified that NVDA reported "The marked second phrase not marked."

### Known issues with pull request:
None with the PR itself.

I did notice that we report "highlighted" for `controlTypes.Role.MARKED_CONTENT`, but we report "marked" for the marked FormatField key, even though these are the same concept. I wonder whether the marked FormatField key should report "highlighted" instead.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Enhanced accessibility in Mozilla Firefox by enabling NVDA to report highlighted text when visiting URLs with text fragments.
- **Bug Fixes**
	- Improved processing of the `formatField` attribute, ensuring accurate handling of the `mark` key for better functionality.

### CODE REVIEW

**Comment by jcsteh** [comment]
Note that Chrome supports text fragments, but even though it scrolls to the fragment, it does not yet expose the highlight to IAccessible2. That change will need to be made in Chrome.

**Comment by XLTechie** [comment]
> I wonder whether the marked FormatField key should report "highlighted" instead.

I would imagine users can more easily relate to the highlighted concept.

**Comment by jcsteh** [comment]
I tend to agree. I wasn't sure whether there was a good reason this is currently reported as "marked", though, so thought I'd avoid making that change here.

**Comment by jcsteh** [comment]
It's not formalised yet. I will file a PR against IA2 at some point to add it. I'm holding off at this point because it's slightly possible (though very unlikely) that I'll discover something before the Firefox patches land that requires me to change how this is exposed.

**Comment by jcsteh** [comment]
https://github.com/LinuxA11y/IAccessible2/pull/30

File: source/NVDAObjects/IAccessible/__init__.py
Comment (jcsteh) [reply source/NVDAObjects/IAccessible/__init__.py:185]: This is inconsistent with the rest of this function and elsewhere, where we don't set keys in format fields unless they deviate from their defaults.

**Review by seanbudd — APPROVED** [review]

**Comment by seanbudd** [comment]
Thanks @jcsteh !

**Pull request merged**
